lib/foreman/renderer/scope/macros/base.rb

```diff
@@ -91,64 +179,157 @@ def default_template_url(template, hostgroup)
+            returns String, desc: 'String describing the kernel release'
+            example 'host_kernel_release(host) # => "3.10.0-957.10.1.el7.x86_6"'
+            example "<%- load_hosts(preload: :kernel_release).each_record do |host| -%>\n<%=   host.name -%>, <%=  host_kernel_release(host) %>\n<%- end -%>"
+          end
           def host_kernel_release(host)
             host&.kernel_release&.value
           end
 
+          apipie :method, 'Returns a host uptime in seconds based on facts' do
+            desc "Given this is based on  e.g. reports from pupept agent, ansible runs or subscription managers
+              the value is only updated on incoming report and can be inaccurate and out of date. An example
+              scenario for such situation is when host reboots but puppet agent hasn't sent new facts yet."
+            required :host, 'Host::Managed', desc: 'host object for which the uptime is returned'
+            returns one_of: [Integer, nil], desc: 'Number representing uptime in seconds or nil in case, there is no information about host boot time'
+            example 'host_uptime_seconds(host) # => 2670619'
+            example 'host_uptime_seconds(host) # => nil'
+          end
           def host_uptime_seconds(host)
             host&.uptime_seconds
           end
 
+          apipie :method, 'Returns amount of memory allocated for the given host' do
+            required :host, 'Host::Managed', desc: 'host object for which the amount is returned'
+            returns Integer, desc: 'amount of memory in megabytes'
+            example 'host_memory(host) # => 11853'
+          end
           def host_memory(host)
             host&.ram
           end
 
+          apipie :method, 'Returns amount of sockets on the given host' do
+            required :host, 'Host::Managed', desc: 'host object for which the amount is returned'
+            returns Integer, desc: 'amount of sockets'
+            example 'host_sockets(host) # => 2'
+          end
           def host_sockets(host)
             host&.sockets
           end
 
+          apipie :method, 'Returns amount of cores on the given host' do
+            required :host, 'Host::Managed', desc: 'host object for which the amount is returned'
+            returns Integer, desc: 'amount of cores'
+            example 'host_cores(host) # => 2'
+          end
           def host_cores(cores)
             host&.cores
           end
 
+          apipie :method, 'Returns true if the given host is virtual, false otherwise' do
+            required :host, 'Host::Managed', desc: "host object to check if it's virtual"
+            returns one_of: [true, false]
+            example 'host_virtual(host) # => true'
+          end
           def host_virtual(host)
             host&.virtual
           end
 
+          apipie :method, 'Compares two package versions' do
+            desc 'It can be used to compare RPM, DEB or gem package versions'
```

Comment:
lzap asked for changing this to "It can be used to compare RPM or DEB package versions but it will not handle edge cases (e.g. 1.0.0.pre1-snapshot-1)."